portal/src/templates.ts

```diff
@@ -1,39 +1,85 @@
+  "templates/{{locale}}/messages/setup_primary_oob_email.html";
 export const TEMPLATE_SETUP_PRIMARY_OOB_EMAIL_TEXT =
-  "templates/__default__/messages/setup_primary_oob_email.txt";
+  "templates/{{locale}}/messages/setup_primary_oob_email.txt";
 export const TEMPLATE_SETUP_PRIMARY_OOB_SMS_TEXT =
-  "templates/__default__/messages/setup_primary_oob_sms.txt";
-export const SetupPrimaryOOBMessageTemplates = [
+  "templates/{{locale}}/messages/setup_primary_oob_sms.txt";
+export const SetupPrimaryOOBMessageTemplatePaths = [
   TEMPLATE_SETUP_PRIMARY_OOB_EMAIL_HTML,
   TEMPLATE_SETUP_PRIMARY_OOB_EMAIL_TEXT,
   TEMPLATE_SETUP_PRIMARY_OOB_SMS_TEXT,
 ] as const;
+export type SetupPrimaryOOBMessageTemplateKeys = typeof SetupPrimaryOOBMessageTemplatePaths[number];
 
 export const TEMPLATE_AUTHENTICATE_PRIMARY_OOB_EMAIL_HTML =
-  "templates/__default__/messages/authenticate_primary_oob_email.html";
+  "templates/{{locale}}/messages/authenticate_primary_oob_email.html";
 export const TEMPLATE_AUTHENTICATE_PRIMARY_OOB_EMAIL_TEXT =
-  "templates/__default__/messages/authenticate_primary_oob_email.txt";
+  "templates/{{locale}}/messages/authenticate_primary_oob_email.txt";
 export const TEMPLATE_AUTHENTICATE_PRIMARY_OOB_SMS_TEXT =
-  "templates/__default__/messages/authenticate_primary_oob_sms.txt";
-export const AuthenticatePrimaryOOBMessageTemplates = [
+  "templates/{{locale}}/messages/authenticate_primary_oob_sms.txt";
+export const AuthenticatePrimaryOOBMessageTemplatePaths = [
   TEMPLATE_AUTHENTICATE_PRIMARY_OOB_EMAIL_HTML,
   TEMPLATE_AUTHENTICATE_PRIMARY_OOB_EMAIL_TEXT,
   TEMPLATE_AUTHENTICATE_PRIMARY_OOB_SMS_TEXT,
 ] as const;
+export type AuthenticatePrimaryOOBMessageTemplateKeys = typeof AuthenticatePrimaryOOBMessageTemplatePaths[number];
 
 export const TEMPLATE_FORGOT_PASSWORD_EMAIL_HTML =
-  "templates/__default__/messages/forgot_password_email.html";
+  "templates/{{locale}}/messages/forgot_password_email.html";
 export const TEMPLATE_FORGOT_PASSWORD_EMAIL_TEXT =
-  "templates/__default__/messages/forgot_password_email.txt";
+  "templates/{{locale}}/messages/forgot_password_email.txt";
 export const TEMPLATE_FORGOT_PASSWORD_SMS_TEXT =
-  "templates/__default__/messages/forgot_password_sms.txt";
-export const ForgotPasswordMessageTemplates = [
+  "templates/{{locale}}/messages/forgot_password_sms.txt";
+export const ForgotPasswordMessageTemplatePaths = [
   TEMPLATE_FORGOT_PASSWORD_EMAIL_HTML,
   TEMPLATE_FORGOT_PASSWORD_EMAIL_TEXT,
   TEMPLATE_FORGOT_PASSWORD_SMS_TEXT,
 ] as const;
+export type ForgotPasswordMessageTemplateKeys = typeof ForgotPasswordMessageTemplatePaths[number];
+
+export type PathTemplate =
+  | SetupPrimaryOOBMessageTemplateKeys
+  | AuthenticatePrimaryOOBMessageTemplateKeys
+  | ForgotPasswordMessageTemplateKeys;
+
+function applyArgumentOnTemplateString(
+  argument: string,
+  template: string,
+  value: string
+): string {
+  return template.replace(new RegExp(`{{[ ]*${argument}[ ]*}}`, "g"), value);
```

Comment:
This substitution looks a bit weird to me. How about making the template types themselves typed? Something like:
```ts
type TemplateType = (locale: string) => string;

const TemplateForgotPasswordEmailHTML: TemplateType = (locale) => `templates/${locale}/messages/forgot_password_email.html`;
```

Reply:
Actually the template is also used for parsing path. Writing template as `(locale: string) => string` will work for rendering the path. Not sure if we should make this symmetric

Reply:
I suppose front-end would need to parse the path to extract list of configured locales? Then, after adding the purposed field to backend, front-end should not need to parse the path.

We can keep it for now, and refactor it after introducing that field.

Reply:
> I suppose front-end would need to parse the path to extract list of configured locales? 

Yes

> Then, after adding the purposed field to backend, front-end should not need to parse the path.

If backend provide the locales, then the whole template thing can be removed and the path can simply be functions `(locale: string) => string`